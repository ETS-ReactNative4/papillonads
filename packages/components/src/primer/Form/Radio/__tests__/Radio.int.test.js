import { renderCustom, renderDefault } from './Radio.int.render'

describe('<Radio />', () => {
  const onChangeMockFn = jest.fn()

  describe('Event', () => {
    test('must handle event when onChange()', () => {
      const mountCustomRender = global.renderMount(renderCustom(onChangeMockFn))
      mountCustomRender.find('input').simulate('change')
      expect(mountCustomRender.props().onChange).toBe(onChangeMockFn)
      expect(mountCustomRender.props().onChange).toHaveBeenCalledWith(expect.any(Object))
    })
  })

  describe('Render', () => {
    test('must match renderDefault()', () => {
      expect(global.renderToJSON(renderDefault())).toMatchSnapshot()
    })
  })
})
