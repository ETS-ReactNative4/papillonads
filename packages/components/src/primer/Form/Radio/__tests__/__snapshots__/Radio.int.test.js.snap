// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> Render must match renderDefault() 1`] = `
Array [
  <div
    className="m-0 form-checkbox"
  >
    <label
      htmlFor="checkbox"
    >
      <input
        checked={false}
        name="fantastic-name"
        onChange={[Function]}
        type="radio"
      />
      White
    </label>
  </div>,
  <div
    className="m-0 form-checkbox"
  >
    <label
      htmlFor="checkbox"
    >
      <input
        checked={true}
        name="fantastic-name"
        onChange={[Function]}
        type="radio"
      />
      Black
    </label>
  </div>,
]
`;
